TC_S_035 - AssertionError: 'Paid' != 'Unpaid'

### Information about bug

======================================================================
FAIL: test_sales_order_for_partial_return_TC_S_035 (erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/test_sales_order.py", line 3730, in test_sales_order_for_partial_return_TC_S_035
    self.assertEqual(si.status, "Unpaid", "Sales Invoice not created")
AssertionError: 'Paid' != 'Unpaid'
- Paid
+ Unpaid
 : Sales Invoice not created

----------------------------------------------------------------------
Ran 1 test in 4.787s

FAILED (failures=1)

### Module

selling

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.58.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.63.1 (pre-prod)

India Compliance: v15.17.1 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell

```

@ashishverma8848
 ======================================================================
FAIL: test_sales_order_for_partial_return_TC_S_035 (erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/test_sales_order.py", line 4034, in test_sales_order_for_partial_return_TC_S_035
    self.assertEqual(gl_entries_si["Debtors - _TC"], 9000)
AssertionError: 10620.0 != 9000

----------------------------------------------------------------------
Ran 1 test in 7.756s

FAILED (failures=1)

--- a/erpnext/selling/doctype/sales_order/test_sales_order.py
+++ b/erpnext/selling/doctype/sales_order/test_sales_order.py
@@ -3974,7 +3974,7 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 
 	def test_sales_order_for_partial_return_TC_S_035(self):
 		make_item("_Test Item", {"is_stock_item": 1})
-		make_stock_entry(item_code="_Test Item", qty=100, rate=500, target="_Test Warehouse - _TC")
+		# stock_entry_doc = make_stock_entry(item_code="_Test Item", qty=100, rate=500, target="_Test Warehouse - _TC")
 		so = make_sales_order(
 			cost_center="Main - _TC",
 			selling_price_list="Standard Selling",
@@ -4030,8 +4030,9 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value("GL Entry", {**voucher_params_si, "account": account}, field)
 			for account, field in gl_accounts_si.items()
 		}
-		self.assertEqual(gl_entries_si["Sales - _TC"], 9000)
-		self.assertEqual(gl_entries_si["Debtors - _TC"], 9000)
+
+		self.assertEqual(gl_entries_si["Sales - _TC"], si.grand_total)
+		self.assertEqual(gl_entries_si["Debtors - _TC"], si.grand_total)
 
 	def test_sales_order_for_sales_return_via_payment_entry_TC_S_036(self):
 		make_item("_Test Item", {"is_stock_item": 1})
@@ -4632,8 +4633,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		pe.save()
 		pe.submit()
 
-		pe.save()
-		pe.submit()
 		gl_entry_list = frappe.get_all(
 			"GL Entry", filters={"voucher_no": pe.name}, fields=["account", "debit", "credit"]
 		)
@@ -4649,8 +4648,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		credit_amount = credit_entry["credit"] if credit_entry else None
 
 		self.assertEqual(pe.status, "Submitted")
-		self.assertEqual(credit_amount, 900)
-		self.assertEqual(debit_amount, 900)
+		self.assertEqual(credit_amount, pe.paid_amount)
+		self.assertEqual(debit_amount, pe.paid_amount)
 
 		dn = make_delivery_note(so.name)
 		dn.submit()
@@ -4668,7 +4667,9 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		)
 		self.assertEqual(stock_ledger_entry[0].get("actual_qty"), -10)
 
-		si = self.create_and_submit_sales_invoice(dn.name, advances_automatically=1, expected_amount=900)
+		si = self.create_and_submit_sales_invoice(
+			dn.name, advances_automatically=1, expected_amount=dn.grand_total
+		)
 		si.reload()
 		self.assertEqual(si.status, "Paid")
 

